tools/crm_mon_print.c

```diff
@@ -54,6 +54,48 @@ static int print_node_attributes(pcmk__output_t *out, pe_working_set_t *data_set
 static int print_failed_actions(pcmk__output_t *out, pe_working_set_t *data_set,
                                 gboolean print_spacer);
 
+static gboolean
+str_in_unames(gchar **unames, const gchar *s) {
+    return unames && g_strv_contains((const gchar * const*) unames, s);
+}
+
+static gchar **
+build_uname_list(pe_working_set_t *data_set, const char *s) {
+    gchar **unames = NULL;
+
+    if (s == NULL) {
+        /* Nothing was given so return a list of all node names. */
+        int i = 0;
+        unames = calloc(g_list_length(data_set->nodes)+1, sizeof(gchar *));
+
+        for (GListPtr iter = data_set->nodes; iter != NULL; iter = iter->next) {
+            pe_node_t *node = (pe_node_t *) iter->data;
+            unames[i] = (gchar *) strdup(node->details->uname);
+            i++;
+        }
+    } else {
+        pe_node_t *node = pe_find_node(data_set->nodes, s);
+
+        if (node) {
+            /* The given string was a valid uname for a node.  Return a
+             * singleton list containing just that uname.
+             */
+            unames = calloc(2, sizeof(gchar *));
+            unames[0] = (gchar *) strdup(s);
```

Comment:
g_strdup((gchar*)...) since we're going to use glib to free it. Nice bonus is it asserts on failure.